packages/starlight/components/LanguageSelect.astro

```diff
@@ -46,4 +46,18 @@ function localizedPathname(locale: string | undefined): string {
 		}
 	}
 	customElements.define('starlight-lang-select', StarlightLanguageSelect);
+	window.addEventListener('pageshow', (event) => {
+		if (!event.persisted) return;
+		document
+			.querySelectorAll<HTMLSelectElement>('starlight-lang-select select')
+			.forEach((select) => {
+				// If the page was loaded from a cache, the language select selected index might not be in
+				// sync with the current page.
+				const markupSelectedIndex =
+					select.querySelector<HTMLOptionElement>('option[selected]')?.index;
+				if (markupSelectedIndex !== select.selectedIndex) {
+					select.selectedIndex = markupSelectedIndex ?? 0;
```

Comment:
```suggestion
					select.selectedIndex = markupSelectedIndex ?? select.selectedIndex;
```

Please correct me if I'm wrong, but wouldn't it be better to leave the dropdown as it is if markupSelectedIndex is undefined?

Or formulated differently: Why did you fallback to the default locale in this case, HiDeoo? 

This is just another option to consider, you are probably right to default to 0...

Reply:
If the selected index from the markup is not defined, we're in the case of the default locale with no `selected` attribute in the markup. For example, if you check the markup on https://starlight.astro.build/, you can see that there is no `selected` attribute for "English" which makes the first option automatically selected.

<img width="577" alt="image" src="https://github.com/user-attachments/assets/05bbad80-e489-4ab9-a8fa-cabbe5426ddc">

Falling back to `0` ensures that the first option is selected in this case even if `select.selectedIndex` has a different value due to a cache restoration.

Basically, when a cache is restored, the markup is always the source of truth and we never fallback to `select.selectedIndex` which may not be reliable.